### PR TITLE
fix: use method name similar to gateway endpoint

### PR DESCRIPTION
**Pull request by dhruvkelawala**
## Motivation and Resolution

...

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes
- [ ] Updated the docs (www)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: __tests__/defaultProvider.test.ts [hunk 1]
```diff
@@ -75,43 +75,41 @@ describe('defaultProvider', () => {
...
-      const classResponse = await testProvider.getClassAt(exampleContractAddress);
+      const classResponse = await testProvider.getClassAt(erc20ContractAddress);
 
       expect(classResponse).toHaveProperty('program');
       expect(classResponse).toHaveProperty('entry_points_by_type');
     });
 
     // TODO see if feasible to split
-    describe('getClassHashAt & GetClass', () => {
+    describe(' GetClass', () => {
```
Comment (ivpavici) [inline __tests__/defaultProvider.test.ts:90]: Maybe:
```suggestion
    describe('GetClassByHash', () => {
```
?

File: src/provider/rpc.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ export class RpcProvider implements ProviderInterface {
     return this.fetchEndpoint('starknet_getTransactionReceipt', { transaction_hash: txHash });
   }
 
-  public async getClass(classHash: RPC.Felt): Promise<RPC.ContractClass> {
+  public async getClassByHash(classHash: RPC.Felt): Promise<RPC.ContractClass> {
```
Comment (tabaktoni) [inline src/provider/rpc.ts:197]: I would recommend creating an alias function the same as we did with getBlock(), and leaving the original name as is.

**Review by tabaktoni — APPROVED** [review]

File: src/provider/rpc.ts (hunk 2 above)
Comment (dhruvkelawala) [reply src/provider/rpc.ts:197]: Just for RPC or for the Provider Interface?

Comment (tabaktoni) [reply src/provider/rpc.ts:198]: I thought only RPC as the endpoint is get_class.
I am not sure how many users we have that will be impacted but in general, I think it is good to do if we can soft transition. where we create alias depercate and than remove in couple of versions in the future.
like:
v1:
methodA()

v2:
@deprecated - use methodB() 
methodA()
...
methodB()

vN:
methodB()

**Pull request merged**